pkg/scheduler/schedule_one.go

```diff
@@ -951,11 +961,15 @@ func (sched *Scheduler) handleSchedulingFailure(ctx context.Context, fwk framewo
 	cachedPod, e := podLister.Pods(pod.Namespace).Get(pod.Name)
 	if e != nil {
 		logger.Info("Pod doesn't exist in informer cache", "pod", klog.KObj(pod), "err", e)
+		// We need to call DonePod here because we don't call AddUnschedulableIfNotPresent in this case.
+		sched.DonePod(pod.UID)
```

Comment:
can we make it via defer and make sure that it's safe to call `DonePod` more than once?